### PR TITLE
migrate checks for rt feed presence

### PR DESCRIPTION
**Pull request by owades**
# Description

Migrate 3 checks for RT feed presence.

Resolves # [issue]

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?
`./dbt test` for each check and downstream
looked at aggregate results to confirm that they look normal

### CODE REVIEW

**Comment by owades** [comment]
@atvaccaro @lauriemerrell , since these are my first RT checks I'd love a detailed review from one of you.

Specific callouts:
- Each check only runs for one of the three feeds, so I filter rt_feed_guideline_index accordingly before joining to it. I also don't run a test checking for equal rows to rt_feed_guideline_index, since it only should contain a subset of rows (ie only one feed)
- Separate from these checks, I modified the tests that @atvaccaro wrote for his first RT check, `no_rt_critical_validation_errors`, please take a look at this as I don't want to cause trouble!

File: warehouse/models/intermediate/gtfs/gtfs_quality/_int_gtfs_quality.yml [hunk 1]
```diff
@@ -12,18 +12,15 @@ models:
...
             - date
             - base64_url
-    columns:
-      - name: status
-        tests:
-          - dbt_utils.not_null_proportion:
-              at_least: 0.99
+            - feed_type
+      - dbt_utils.equal_rowcount:
+          compare_model: ref('int_gtfs_quality__schedule_feed_guideline_index')
```
Comment (lauriemerrell) [inline warehouse/models/intermediate/gtfs/gtfs_quality/_int_gtfs_quality.yml]: I don't think that this is necessarily expected to have an equal row count... We expect to have a moderately different number of schedule and RT feeds, I think?

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_present_service_alerts.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
+WITH feed_guideline_index AS (
+    SELECT * FROM {{ ref('int_gtfs_quality__rt_feed_guideline_index') }}
+     WHERE feed_type = 'service_alerts'
+),
+
+fct_daily_rt_feed_files AS (
```
Comment (lauriemerrell) [inline warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_present_service_alerts.sql:6]: Sorry to keep doing this (nuance/confusion about indexing on feeds/files vs. URLs/dataset records in Airtable), but in #2010 I have a daily URLs list for RT which might be better than checking files -- daily URLs will say whether we were *attempting* to hit a service alerts feed. Issue with `fct_daily_rt_feed_files` is that feeds with 0 files will not be listed, but if this is just checking for existence of a feed I think that the list of URLs is better.

Comment (lauriemerrell) [reply warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_present_service_alerts.sql:6]: But defer to you if you only want to give credit for feeds where we can get at least one file, I think that could make sense; just not sure where this check is meant to fall on the spectrum of success/validity vs. pure existence.

**Review by lauriemerrell — COMMENTED** [review]
Couple of questions

File: warehouse/models/intermediate/gtfs/gtfs_quality/_int_gtfs_quality.yml [hunk 3]
```diff
@@ -12,18 +12,15 @@ models:
             - date
 
   - name: int_gtfs_quality__no_rt_critical_validation_errors
-    description: |
-      Checks for any critical validation notices in RT feeds.
-    tests:
+    tests: &stg_gtfs_guideline_tests_rt
       - dbt_utils.unique_combination_of_columns:
```
[the remark this answers is not in the document]
Comment (owades) [reply warehouse/models/intermediate/gtfs/gtfs_quality/_int_gtfs_quality.yml]: Nope, was just trying to be thorough. Thanks for explaining, makes a lot of sense.

**Review by lauriemerrell — APPROVED** [review]
Ok to merge with the current logic, just want to note that I think it's possible we'll want to revisit the indexing here if we ever want to change this to a "URL exists" check as opposed to a "files exist" check.

**Pull request merged**